src/Adapter/Alias/QueryHandler/SearchAliasForAssociationHandler.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Alias\QueryHandler;
+
+use PrestaShop\PrestaShop\Adapter\Alias\Repository\AliasRepository;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Query\SearchAliasesForAssociation;
+use PrestaShop\PrestaShop\Core\Domain\Alias\QueryHandler\SearchAliasesForAssociationHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Alias\QueryResult\AliasForAssociation;
+
+class SearchAliasForAssociationHandler implements SearchAliasesForAssociationHandlerInterface
+{
+    public function __construct(private readonly AliasRepository $aliasRepository)
```

Comment:
The `protected` makes sense indeed, I thought readonly would work even if e extend because even in the child class it's only set once But I don't know enough the details of property promotion so I trust you, let's be safe on this one

Reply:
I've tried it  again now in sandbox with simple class inheritence and it seems to still work with parent construct and readonly (but im pretty sure i had some issues with in in the past, I'll try to find where and what caused this idea exactly)

Reply:
yeah so, after all lets leave them readonly. I cannot reproduce the issue I had, so I tend to think I confused something, as logically we don't want to allow those services to be reset after container is built :smile: srry for confusion